plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityBoatType.java

```diff
@@ -1,85 +1,70 @@
 package com.denizenscript.denizen.objects.properties.entity;
 
+import com.denizenscript.denizen.nms.NMSHandler;
+import com.denizenscript.denizen.nms.NMSVersion;
 import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.utilities.BukkitImplDeprecations;
 import com.denizenscript.denizencore.objects.core.ElementTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
-import com.denizenscript.denizencore.objects.properties.Property;
 import com.denizenscript.denizencore.objects.properties.PropertyParser;
 import org.bukkit.TreeSpecies;
 import org.bukkit.entity.Boat;
 
-public class EntityBoatType implements Property {
+public class EntityBoatType extends EntityProperty<ElementTag> {
 
-    public static boolean describes(ObjectTag object) {
-        return object instanceof EntityTag
-                && ((EntityTag) object).getBukkitEntity() instanceof Boat;
-    }
-
-    public static EntityBoatType getFrom(ObjectTag object) {
-        if (!describes(object)) {
-            return null;
-        }
-        else {
-            return new EntityBoatType((EntityTag) object);
-        }
-    }
-
-    public static final String[] handledMechs = new String[] {
-            "boat_type"
-    };
+    // <--[property]
+    // @object EntityTag
+    // @name boat_type
+    // @input ElementTag
+    // @description
+    // Controls the wood type of the boat.
+    // Valid wood types can be found here: <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/TreeSpecies.html>
+    // Deprecated in versions 1.19 and above. Use <@link property EntityTag.color>.
+    // -->
 
-    public EntityBoatType(EntityTag entity) {
-        this.entity = entity;
+    public static boolean describes(EntityTag boat) {
+        return boat.getBukkitEntity() instanceof Boat;
     }
 
-    EntityTag entity;
-
     @Override
-    public String getPropertyString() {
-        return getBoat().getWoodType().name();
+    public ElementTag getPropertyValue() {
+        return getBoatType();
     }
 
     @Override
     public String getPropertyId() {
         return "boat_type";
     }
 
-    public Boat getBoat() {
-        return (Boat) entity.getBukkitEntity();
+    @Override
+    public void setPropertyValue(ElementTag type, Mechanism mechanism) {
+        setBoatType(type);
     }
 
-    public static void register() {
+    public ElementTag getBoatType() {
+        return new ElementTag(as(Boat.class).getWoodType());

```

Comment:
Can't this method be removed and the impl be put directly into getPropertyValue?